### ISSUE TITLE
Improve the news slug

### ISSUE THREAD

**Issue by wenche** [equinor/energyvision#652]
### Background

The news article was the first template we created in this project. Because news is published at regular intervals and not revisited like the topic pages, we took a different approach and decided to keep news outside of the Route concept. Bacause of that decision we have the slug field added directly in the news template. 

However, we improved the way we handled the slugs when we worked on the topic pages. We wanted to have the complete URL in the slugs field (or the part after the lang part) https://equinor.com/no`/karriere/min-karriere`. The upside is that Sanity is the single source of truth and we avoid messy logic in the web. I also make it easy to have the dynamic catch-all slug approach.  We should update the news template to use this approach as well, and this will probably make things less complicated for the satellite pages as well. Win - win.

### TODO
- Update the slugs field in the news article to use the same logic as in the routes: One text field for the custom part for this news article ("Equinor increases its ownership in the Statfjord field") and a corresponding "Generate slug" button.
- Find a way to let Sanity know what to prefix the news article with (en-GB: "news", nb-NO: "nyheter") that also makes sense with the satellite pages. See comment about preview by id at the bottom
- Remove the logic related to the news template queries in the web?
- Hopefully remove the separate news article in /pages/news/[slug] and use the catch all route with a page template instead

Related: #308 to improve the mechanism for slugs fields

### Acceptance criteria
- Publisher can add a text in plain English or Norwegian that will express the URL (`What we do about the environment/Hva vi gjør for miljøet`)
- By pushing  a button the plain text is slugified and  the complete slug including parent folder is visible (`/sustainability/what-we-do-about-the-environment` | `/barekraft/hva-vi-gjor-for-miljoet`
- The publisher can NOT edit the final generated slug directly
- The final slug field should still be of type `slug`

**Comment by wenche** [comment]
Should have a meeting for this before we start to make sure that everybody knows what to do

**Comment by wenche** [comment]
Connected to #308

**Comment by wenche** [comment]
The preview for news should probably work in the same way as with the rest of the pages: If there's no slug, preview by id instead. Currently the preview will show a message: "The product (!) needs a slug before it can be previewed." We should fix this, since actually adding the slug should not be the first thing to do. 

I _think_ we are able to preview by both id and slug, even if everything is in one document? If not, let's consider  to just preview by ids all the time. This might also actually be easier, since the id has language information attached for the translations `b2f7b874-7b05-4a4f-8033-f4b26841894e__i18n_nb_NO`

**Comment by nilsml** [comment]
@adneforslund to document why we want to change the news slug, benefits vs drawbacks

**Comment by adneforslund** [comment]
Add generate button, but not add routesm dont add dates with generate

**Comment by BirteThornquist** [comment]
@fernandolucchesi I have updated the UX for creating slugs in Figma [here ](https://www.figma.com/file/F2OnKFEoMwFaANMgeGRGqE/CMS?node-id=4599%3A81487)

Moved the generate button, removed the field for the generated slug to just a write out of the slug and changed the help text. 

![Skjermbilde 2022-03-03 kl. 09.13.52.png](https://images.zenhubusercontent.com/5fb7acdfad906a977e792e56/24841b9d-4d2c-4b10-9e92-18e6557784d3)

![Skjermbilde 2022-03-03 kl. 09.15.42.png](https://images.zenhubusercontent.com/5fb7acdfad906a977e792e56/e07536b2-798b-4790-9c10-51da74dcd6b1)

**Comment by fernandolucchesi** [comment]
This is ready for review, could you take a look @wenche?

**Comment by wenche** [comment]
Sanity:

![Skjermbilde 2022-03-10 kl. 09.59.53.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/51b71f4c-53ff-49ae-b985-91f2b0d419c0)

![Skjermbilde 2022-03-10 kl. 10.13.34.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/141f7193-9c8c-4271-a3c4-564897b11297)
![Skjermbilde 2022-03-10 kl. 10.17.28.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/dfcc9457-1942-4d4f-b616-ba71c7e52499)
![Skjermbilde 2022-03-10 kl. 10.19.34.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/584c9d23-9f9b-4f95-a990-d411f21a9688)

- [ ] Can't edit a slug
- [ ] News slugs are prefixed with corresponding name of the news folder for the chosen language

**Comment by wenche** [comment]
The design will be improved according to the sketches by #308

This is a HUGE change so it's not that easy to test 100%. It looks good at first glance, but we just have to keep our eyes open to see if something is missing or isn't correct :thumbsup:
